gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -135,14 +147,18 @@ public String toString() {
     }
   }
 
-  private BlockingQueue<Dag<JobExecutionPlan>> queue;
+  private BlockingQueue<Dag<JobExecutionPlan>>[] queue;
+  private BlockingQueue<String>[] cancelQueue;
```

Comment:
Do you still need blockingQueue ?

Reply:
yes, separate queues for separate DagManagerThreads